symfit/core/objectives.py

```diff
@@ -57,50 +57,84 @@ def sigma_data(self):
             self.model)
 
     @abc.abstractmethod
-    def __call__(self, **parameters):
+    def __call__(self, ordered_parameters=[], **parameters):
         """
         Evaluate the objective function for given parameter values.
 
-        :param parameters:
-        :return: float
+        :param ordered_parameters: List of parameter, in alphabetical order.
+            Typically provided by the minimizer.
+        :param parameters: parameters as keyword arguments.
+        :return: evaluated model.
         """
-        pass
+        # zip will stop when the shortest of the two is exhausted
+        parameters.update(dict(zip(self.model.unfixed_params, ordered_parameters)))
+        parameters.update(self.invariant_kwargs)
+        return self.model(**key2str(parameters))
+
+    @cached_property
+    def invariant_kwargs(self):
+        """
+        Prepares the invariant kwargs to ``self.model`` which are not provided
+        by the minimizers, and are the same for every iteration of the
+        minimization. This means fixed parameters and data, matching the
+        signature of ``self.model``.
+        """
+        kwargs = {p: p.value for p in self.model.params
+                     if p not in self.model.unfixed_params}
+        data_by_name = key2str(self.data)
+        kwargs.update(
+            {p: data_by_name[p] for p in
+            self.model.__signature__.parameters if p in data_by_name}
+        )
+        return kwargs
+
+    def _clear_cache(self):
+        """
+        Clear cached properties. Should preferably be called after every
+        minimization precedure to clean the pallette.
+        """
+        del self.invariant_kwargs
```

Comment:
This should be a general method on one of the most basic classes that wipes *all* cached properties. Also, did you test this? It looks funny to me.

Reply:
I'm still looking for a good place to call this after any execute, so suggestions are welcome.

The `del` indeed looks weird but it is correct, when you call del on a `cached_property` it is overloaded to delete the associated cache, not the whole property.